skyportal/handlers/api/gcn.py

```diff
@@ -900,6 +1015,69 @@ def add_tiles(localization_id):
         Session.remove()
 
 
+def add_skymap_properties(localization_id, user_id):
+    session = Session()
+    try:
+        localization = session.scalar(
+            sa.select(Localization).where(Localization.id == localization_id)
+        )
+        user = session.scalar(sa.select(User).where(User.id == user_id))
+        sky_map = localization.table
+
+        properties_dict = {}
+        tags_list = []
+        result = crossmatch(sky_map, contours=(0.9,), areas=(500,))
+        area = result.contour_areas[0]
+        prob = result.area_probs[0]
+
+        if not np.isnan(area):
+            properties_dict["area_90"] = area
+            if properties_dict["area_90"] < 500:
+                tags_list.append("< 500 sq. deg.")
+        if not np.isnan(prob):
+            properties_dict["probability_500"] = prob
+            if properties_dict["probability_500"] >= 0.9:
+                tags_list.append("> 0.9 in 500 sq. deg.")
+
+        # Distance stats
+        if 'DISTMU' in sky_map.dtype.names:
+            # Calculate the cumulative area in deg2 and the cumulative probability.
+            dA = moc.uniq2pixarea(sky_map['UNIQ'])
+            dP = sky_map['PROBDENSITY'] * dA
+            mu = sky_map['DISTMU']
+            sigma = sky_map['DISTSIGMA']
+
+            distmean, _ = distance.parameters_to_marginal_moments(dP, mu, sigma)
+            if not np.isnan(distmean):
+                properties_dict["distance"] = distmean
+                if distmean <= 150:
+                    tags_list.append("< 150 Mpc")
+
+        properties = LocalizationProperty(
+            localization_id=localization_id, sent_by_id=user.id, data=properties_dict
+        )
+        session.add(properties)
+
+        tags = [
+            LocalizationTag(
+                localization_id=localization_id,
+                text=text,
+                sent_by_id=user.id,
+            )
+            for text in tags_list
+        ]
+        session.add_all(tags)
+
+        session.commit()
+        return log(f"Generated properties for localization {localization_id}")
+    except Exception as e:
+        return log(
+            f"Unable to generate properties for localization {localization_id}: {e}"
+        )
+    finally:
+        Session.remove()
```

Comment:
Why `remove`?

Reply:
That's what most of our IOloop sessions call.